File: 1431. Kids With the Greatest Number of Candies/main.go
package main

import "fmt"

func kidsWithCandies(candies []int, extraCandies int) []bool {
	res := make([]bool, 0, len(candies))
	maxcand := 0
	for _, value := range candies {
		if value > maxcand {
			maxcand = value
		}
	}
	for i := range len(candies) {
		if candies[i]+extraCandies >= maxcand {
			res = append(res, true)
		} else {
			res = append(res, false)
		}
	}
	return res
}
func main() {
	candies := []int{2, 3, 5, 1, 3}
	extraCandies := 3
	result := kidsWithCandies(candies, extraCandies)
	fmt.Println(result) // Вывод: [true true true false true]
}

File: 1431. Kids With the Greatest Number of Candies/main.py
from typing import List
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        res = []
        maxcand = max(candies)
        for i in candies:
            if i + extraCandies >= maxcand:
                res.append(True)
            else:
                res.append(False)
        return res
    
solution = Solution()

# Тестовые случаи
test_cases = [
    ([2, 3, 5, 1, 3], 3),  # Ожидаемый результат: [True, True, True, False, True]
    ([4, 2, 1, 1, 2], 1),  # Ожидаемый результат: [True, False, False, False, False]
    ([12, 1, 12], 10),     # Ожидаемый результат: [True, False, True]
]

# Проверяем каждый тестовый случай
for candies, extra in test_cases:
    result = solution.kidsWithCandies(candies, extra)
    print(f'candies: {candies}, extraCandies: {extra} => result: {result}')